src/ast/mod.rs

```diff
@@ -1004,6 +1005,51 @@ pub enum OneOrManyWithParens<T> {
     Many(Vec<T>),
 }
 
+impl<T> Deref for OneOrManyWithParens<T> {
+    type Target = [T];
+
+    fn deref(&self) -> &[T] {
+        match self {
+            OneOrManyWithParens::One(one) => core::slice::from_ref(one),
+            OneOrManyWithParens::Many(many) => many,
+        }
+    }
+}
+
+impl<T> AsRef<[T]> for OneOrManyWithParens<T> {
+    fn as_ref(&self) -> &[T] {
+        self
+    }
+}
+
+impl<'a, T> IntoIterator for &'a OneOrManyWithParens<T> {
+    type Item = &'a T;
+    type IntoIter = core::slice::Iter<'a, T>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.iter()
+    }
+}
+
+impl<T> IntoIterator for OneOrManyWithParens<T> {
+    type Item = T;
+
+    #[cfg(not(feature = "std"))]
+    type IntoIter = core::iter::Chain<core::option::IntoIter<T>, alloc::vec::IntoIter<T>>;
+
+    #[cfg(feature = "std")]
+    type IntoIter = core::iter::Chain<core::option::IntoIter<T>, std::vec::IntoIter<T>>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        let (one, many) = match self {
+            OneOrManyWithParens::One(one) => (Some(one), Vec::new()),
+            OneOrManyWithParens::Many(many) => (None, many),
+        };
+
+        one.into_iter().chain(many)
```

Comment:
This is a bit strange, it could be 
1) `vec![one]` and just return `Vec::into_iter`, but it allocates, or
2) create and implement a new iterator, but it's much more work, and we won't get for free additional iterator traits impls.

Just let me know if other approach is preferable

Reply:
> create and implement a new iterator, but it's much more work, and we won't get for free additional iterator traits impls.

this is what i would have expected. 

Not sure what you mean "won't get for free additional iterator traits impl"

Another potential approach is returning a `Box`ed iterator

Reply:
It wasn't too much work after all :grimacing: 

> Not sure what you mean "won't get for free additional iterator traits impl"

It's actually just `DoubleEndedIterator`, I thought there were more